### PR TITLE
Fix issue 966 dot array in old() function

### PR DESCRIPTION
**Pull request by gabpnr**
Fix issue 966 dot array in old() function

Signed-off-by: Gabriel P <16049869+gabpnr@users.noreply.github.com>

### CODE REVIEW

File: system/HTTP/IncomingRequest.php [hunk 1]
```diff
@@ -490,6 +490,20 @@ public function getOldInput(string $key)
 		{
 			return $_SESSION['_ci_old_input']['get'][$key];
 		}
+
+		helper('array');
+
+		// Check for an array value in POST.
+		if (isset($_SESSION['_ci_old_input']['post']) && dot_array_search($key, $_SESSION['_ci_old_input']['post']))
```
Comment (lonnieezell) [inline system/HTTP/IncomingRequest.php]: You're doing the dot array search twice here which is inefficient. More inefficient than assigning it to a variable, I believe. Please change that.

File: system/HTTP/IncomingRequest.php [hunk 2]
```diff
@@ -490,6 +490,20 @@ public function getOldInput(string $key)
...
+		{
+			return dot_array_search($key, $_SESSION['_ci_old_input']['post']);
+		}
+
+		// Check for an array value in GET.
+		if (isset($_SESSION['_ci_old_input']['get']) && dot_array_search($key, $_SESSION['_ci_old_input']['get']))
```
Comment (lonnieezell) [inline system/HTTP/IncomingRequest.php]: Make changes here, also.

**Comment by gabpnr** [comment]
hum, the coverage check is ok in my fork ...

**Comment by lonnieezell** [comment]
@gabpnr The coverage calculations are a little weird sometimes. Your code is fine. Merging. Thanks!

**Pull request merged**